zonemaster/zonemaster-engine
# Update unit tests for Consistency05 and Consistency06

## Purpose

The PR creates unit tests based on the new test zone framework. The unit tests in this PR is based on the data in https://github.com/zonemaster/zonemaster/pull/1213. It also removes old unit tests.

Due to a bug in the implementation of Constistency05 and Consistency06, respectively, some scenarios cannot presently be used. The following scenarios have not been included as unit tests due to the bug:

* CHILD-ZONE-LAME-1 (Consistency05)
* IB-ADDR-MISMATCH-3 (Consistency05)
* ONE-SOA-MNAME-4 (Consistency06)
* NO-RESPONSE (Consistency06)

The bug is described in #1300 and #1301.

## Changes

New unit tests are added in `t/Test-consistency05.t` and `t/Test-consistency06.t` (plus data files).

Old unit tests for Consistency05 and Consistency06 are removed, except for `t/Test-consistency05-F.t` and `t/Test-consistency06-B.t` that have been kept since they test for the presence of `NO_RESPONSE` (with the help of "fake delegation").

The `MANIFEST` file has also been updated.

## How to test this PR

Review https://github.com/zonemaster/zonemaster/pull/1213 and verify that the CI passes.
